esrally/utils/git.py

```diff
@@ -22,6 +22,11 @@
 from esrally.utils import io, process
 
 
+def with_retry(command):
+    # git has no native concept of retries
+    return f'/bin/bash -c "for i in {{1..5}}; do {command} && s=0 && break || s=1 && sleep 1; done; (exit $s)"'
```

Comment:
Given we now invoke real git operations in the test (including cloning, fetching, etc.), I figured it'd be a prerequisite to ensure that these commands were naively retryable.

Reply:
Reverted in https://github.com/elastic/rally/pull/1602/commits/59b762e29b55d6262be9dd5b32b0b2d0d13b6296